### PR TITLE
DOCS-1520: Document App Client new methods 

### PR DESCRIPTION
**Pull request by sguequierre**
Add documentation for the 7 methods that were added to Cloud App API

### CODE REVIEW

File: docs/build/program/apis/cloud.md [hunk 1]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Create an organization invite and send it by email.
+
+**Parameters:**
+
+- `email` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The email address to send the invite to.
+- `authorizations` [(Optional[List[viam.proto.app.Authorization]])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.Authorization): Specifications of the
+  authorizations to include in the invite.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1347]:
```suggestion
- `authorizations` [(Optional[List[viam.proto.app.Authorization]])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.Authorization): Specifications of the authorizations to include in the invite.
```
One line I think?

Also, can you add an example or two of what a "specification" would be? Or is that not practical and the user should just click the link for more info (fine if that's the case)?

File: docs/build/program/apis/cloud.md [hunk 2]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### ListKeys
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+List all keys for the currently-authed-to org.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1576]:
```suggestion
List all keys for the {{< glossary_tooltip term_id="organization" text="organization" >}} that you are currently authenticated to.
```

File: docs/build/program/apis/cloud.md [hunk 3]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
 
 {{% /tab %}}
 {{< /tabs >}}
+
+### CreateOrganizationInvite
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Create an organization invite and send it by email.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1341]:
```suggestion
Create an {{< glossary_tooltip term_id="organization" text="organization" >}} invite and send it by email.
```
Consider adding glossary defs to Viam-specific terms in every description

File: docs/build/program/apis/cloud.md [hunk 4]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+- `email` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The email address associated with the invite.
+
+**Returns:**
+
+- None.
+
+**Raises:**
+
+- `GRPCError`: If no pending invite is associated with the provided email address.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1439]:
```suggestion
- `GRPCError`: This error is raised if no pending invite is associated with the provided email address.
```

File: docs/build/program/apis/cloud.md [hunk 5]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### GetRoverRentalRobots
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Return a list of rover rental robots within an org.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1455]:
```suggestion
Return a list of rover rental robots within an {{< glossary_tooltip term_id="organization" text="organization" >}}.
```

File: docs/build/program/apis/cloud.md [hunk 6]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+### CheckPermissions
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions to validate. For example, `“read_organization”` or `“control_robot”`.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1483]:
```suggestion
- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions to validate.
  For example, `“read_organization”` or `“control_robot”`.
```

File: docs/build/program/apis/cloud.md [hunk 7]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### CheckPermissions
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1479]: Can you include a quick description of what validity means?

File: docs/build/program/apis/cloud.md [hunk 8]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions to validate. For example, `“read_organization”` or `“control_robot”`.
+  See [GitHub](https://github.com/viamrobotics/app/blob/7ddb29e4dcb8cbca6c76c3acaeeb19c544028bd2/auth/models/models.go#L190) for the full list of permissions.
+
+**Returns:**
+
+- [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions argument, with invalid permissions filtered out.
+
+**Raises:**
+
+- `GRPCError`: If the list of permissions to validate is empty.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1492]:
```suggestion
- `GRPCError`: This error is raised if the list of permissions to validate is empty.
```

File: docs/build/program/apis/cloud.md [hunk 9]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### CreateKey
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Create a new API key.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1514]: Would it make sense to link to a relevant part of the docs about API keys?

File: docs/build/program/apis/cloud.md [hunk 10]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+- `authorizations` [(List[APIKeyAuthorization])](https://python.viam.dev/autoapi/viam/app/app_client/index.html#viam.app.app_client.APIKeyAuthorization): A list of authorizations to associate with the key.
+- `name` [(Optional[str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): A name for the key. If None, defaults to the current timestamp.
+
+**Returns:**
+
+- [(Tuple[str, str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The api key and api key ID.
+
+**Raises:**
+
+- `GRPCError`: If the authorizations list is empty.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1527]:
```suggestion
- `GRPCError`: This error is raised if the authorizations list is empty.
```

File: docs/build/program/apis/cloud.md [hunk 11]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+Create a new API key with an existing key’s authorizations.
+
+**Parameters:**
+
+- `id` [(str)](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The ID of the API key to duplicate authorizations from.
+
+**Returns:**
+
+- [(Tuple[str, str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The new api key and api key ID.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1559]:
```suggestion
- [(Tuple[str, str])](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str): The new API key and API key ID.
```

File: docs/build/program/apis/cloud.md [hunk 12]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### DeleteOrganizationInvite
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Delete a pending organization invite.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1399]: [opt] Would it make sense to say
```suggestion
Delete a pending organization invite to the organization you are currently authenticated to.
```

File: static/include/services/apis/cloud.md [hunk 13]
```diff
@@ -43,3 +43,12 @@ Method Name | Description
...
 [`ListModules`](/build/program/apis/cloud/#listmodules ) | List available modules.
+[`CreateOrganizationInvite`](/build/program/apis/cloud/#createorganizationinvite) | Create an organization invite and send it by email.
+[`DeleteOrganizationMember`](/build/program/apis/cloud/#deleteorganizationmember) | Remove a member from the organization.
+[`DeleteOrganizationInvite`](/build/program/apis/cloud/#deleteorganizationinvite) | Delete a pending organization invite.
+[`ResendOrganizationInvite`](/build/program/apis/cloud/#resendorganizationinvite) | Resend a pending organization invite email.
+[`GetRoverRentalRobots`](/build/program/apis/cloud/#getroverrentalrobots) | Return a list of rover rental robots within an org.
+[`CheckPermissions`](/build/program/apis/cloud/#checkpermissions) | Check validity of a list of permissions.
+[`CreateKey`](/build/program/apis/cloud/#createkey) | Create a new API key.
+[`CreateKeyFromExistingKeyAuthorizations`](/build/program/apis/cloud/#createkeyfromexistingkeyauthorizations) | Create a new API key with an existing key’s authorizations.
+[`ListKeys`](/build/program/apis/cloud/#listkeys) | List all keys for the currently-authed-to org.
```
Comment (JessamyT) [inline static/include/services/apis/cloud.md:54]:
```suggestion
[`ListKeys`](/build/program/apis/cloud/#listkeys) | List all keys for an organization.
```
Let's take out the awkward phrasing and let them see the details when the go to the full method description.

File: docs/build/program/apis/cloud.md (hunk 9 above)
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1514]: I can't find one besides the CLI. I think the update to RBAC I'm working on could be a good place to link to in the future.

File: docs/build/program/apis/cloud.md [hunk 14]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### DeleteOrganizationMember
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Remove a member from the organization.
```
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1372]: @stuqdog Is this accurate?

File: docs/build/program/apis/cloud.md (hunk 1 above)
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1347]: It just means those objects. I'll remove the specifications phrasing

File: docs/build/program/apis/cloud.md [hunk 15]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### CheckPermissions
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
```
Comment (sguequierre) [inline docs/build/program/apis/cloud.md:1478]: I assume this means they are authorized for a resource? @stuqdog Can you confirm

File: docs/build/program/apis/cloud.md [hunk 16]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions to validate.
```
Comment (sguequierre) [inline docs/build/program/apis/cloud.md:1482]:
```suggestion
- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions to validate as authorized for a resource.
```

File: docs/build/program/apis/cloud.md (hunk 15 above)
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1478]: Added that below

File: docs/build/program/apis/cloud.md [hunk 17]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): A list of the resource permissions arguments to validate as authorized.
+  For example, `“read_organization”` or `“control_robot”`.
+  See [GitHub](https://github.com/viamrobotics/app/blob/7ddb29e4dcb8cbca6c76c3acaeeb19c544028bd2/auth/models/models.go#L190) for the full list of permissions.
```
Comment (sguequierre) [inline docs/build/program/apis/cloud.md:1484]:
```suggestion
  See [GitHub](https://github.com/viamrobotics/app/blob/main/auth/models/models.go#L190) for the full list of permissions.
```

File: docs/build/program/apis/cloud.md (hunk 7 above)
Comment (JessamyT) [reply docs/build/program/apis/cloud.md:1478]: How about linking to the permissions docs as well for context?

File: docs/build/program/apis/cloud.md [hunk 18]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+### CheckPermissions
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): A list of the resource permissions arguments to validate as authorized.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1482]: I find this hard to parse. How about 
```suggestion
- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): A list containing the resource type and ID, followed by the resource permissions to check.
```

File: docs/build/program/apis/cloud.md [hunk 19]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+  For example, `“read_organization”` or `“control_robot”`.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1483]:
```suggestion
```
Since this example doesn't contain the resource type and ID, maybe take it out and instead let them refer to the example below, and add a comment to that example.

File: docs/build/program/apis/cloud.md [hunk 20]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+**Raises:**
+
+- `GRPCError`: This error is raised if the list of permissions to validate is empty.
+
+```python {class="line-numbers linkable-line-numbers"}
+from viam.proto.app import AuthorizedPermissions
+
+permissions = [AuthorizedPermissions(resource_type="organization",
+                                     resource_id="organization-identifier123",
+                                     permissions=["control_robot"])]
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1499]:
```suggestion
                                     permissions=["control_robot", "read_robot_logs"])]
```
To make it clear they can pass a list

File: docs/build/program/apis/cloud.md [hunk 21]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): A list of the resource permissions arguments to validate as authorized.
+  For example, `“read_organization”` or `“control_robot”`.
+  See [GitHub](https://github.com/viamrobotics/app/blob/main/auth/models/models.go#L190) for the full list of permissions.
+
+**Returns:**
+
+- [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions argument, with invalid permissions filtered out.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1488]: Hmm I'm realizing I'm not sure what this method does. @stuqdog ?
```suggestion
- [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions argument, with invalid permissions filtered out.
For example, if you pass a location with its ID, followed by permissions `"read_fragment"` and `"write_fragment"`, and the location can read but not write fragments in the organization you are currently authenticated to, the method will return `["read_fragment"]`.
```
or is it
```suggestion
- [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): The permissions argument, with invalid permissions filtered out.
For example, if you pass a location with its ID, followed by permissions `"read_fragment"` and `"write_fragment"`, and the organization you are currently authenticated to can read but not write fragments in the specified location, the method will return `["read_fragment"]`.
```
or none of the above? Sorry if these are bad examples--not sure how it works!

File: docs/build/program/apis/cloud.md [hunk 22]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+**Raises:**
+
+- `GRPCError`: This error is raised if the list of permissions to validate is empty.
+
+```python {class="line-numbers linkable-line-numbers"}
+from viam.proto.app import AuthorizedPermissions
+
+permissions = [AuthorizedPermissions(resource_type="organization",
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1497]:
```suggestion
# Check whether organization of ID "organization-identifier123" has permissions to control robots and/or read robot logs within the organization you are currently authenticated to
permissions = [AuthorizedPermissions(resource_type="organization",
```
Or whatever is true...I think this method is confusing and it'd be good to clarify

**Review by JessamyT — COMMENTED** [review]
Need more clarity on the permissions method--otherwise looks good

File: docs/build/program/apis/cloud.md [hunk 23]
```diff
@@ -1332,3 +1332,262 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check validity of a list of permissions.
+
+**Parameters:**
+
+- `permissions` [(List[viam.proto.app.AuthorizedPermissions])](https://python.viam.dev/autoapi/viam/proto/app/index.html#viam.proto.app.AuthorizedPermissions): A list of the resource permissions arguments to validate as authorized.
+  For example, `“read_organization”` or `“control_robot”`.
+  See [GitHub](https://github.com/viamrobotics/app/blob/main/auth/models/models.go#L190) for the full list of permissions.
```
Comment (sguequierre) [inline docs/build/program/apis/cloud.md:1484]: @JessamyT Do you have any idea why this GitHub link keeps on failing htmltest? Idk what is wrong

Comment (JessamyT) [reply docs/build/program/apis/cloud.md:1484]: Hmm I'm not sure--have you re-run it multiple times?

Comment (npentrel) [reply docs/build/program/apis/cloud.md:1484]: Because it's not public code. You should not link to app code - is there something else you can link to?

File: docs/build/program/apis/cloud.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (stuqdog) [reply docs/build/program/apis/cloud.md:1371]: That is correct, at least for now! Everything is subject to change once we have user-level auth in the python SDK.

File: docs/build/program/apis/cloud.md (hunk 15 above)
Comment (stuqdog) [reply docs/build/program/apis/cloud.md:1478]: Yeah so `check_permissions` takes a list of permissions and returns a filtered list of which ones are actual, so e.g., if you have permission to `control_robot` and you pass a `control_robot` permission then you will get that permission back. If you _don't_ have `control_robot` permission and you pass it as your argument, you will get nothing back.

File: docs/build/program/apis/cloud.md (hunk 21 above)
Comment (stuqdog) [reply docs/build/program/apis/cloud.md:1488]: I _believe_ it's the former. You pass some sort of ID (org, user, location, etc.) and then it returns which permissions that ID has access to. I suspect fleet knows more. @Kschappacher you were the last one to touch the `AuthorizedPermissions` type in app.proto (admittedly many months ago), can you confirm?

**Review by stuqdog — APPROVED** [review]
looks good to me, modulo existing comments.

File: docs/build/program/apis/cloud.md (hunk 21 above)
Comment (Kschappacher) [reply docs/build/program/apis/cloud.md:1488]: Yupp that is correct. You can pass in any kind of resource id along with the permission you want to check and it will return subset of permissions the resource  actually has access to.

File: docs/build/program/apis/cloud.md (hunk 22 above)
Comment (JessamyT) [reply docs/build/program/apis/cloud.md:1497]:
```suggestion
# Check whether the entity you're currently authenticated to has permission to control and/or read logs from robots in the organization-identifier123 org
permissions = [AuthorizedPermissions(resource_type="organization",
```
Based on offline convo I think it's actually this?

File: docs/build/program/apis/cloud.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1371]: @stuqdog Thanks!

File: docs/build/program/apis/cloud.md (hunk 22 above)
Comment (sguequierre) [reply docs/build/program/apis/cloud.md:1497]: Awesome, thanks!

File: docs/build/program/apis/cloud.md [hunk 24]
```diff
@@ -1332,3 +1332,323 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+
+{{% /tab %}}
+{{< /tabs >}}
+
+### CheckPermissions
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Check if the organization, location, or robot your `ViamClient` is authenticated to is permitted to perform some action or set of actions on the resource you pass to the method.
```
Comment (JessamyT) [inline docs/build/program/apis/cloud.md:1478]: Love this wording--much more descriptive, nice.

**Review by JessamyT — APPROVED** [review]
LGTM

File: docs/build/program/apis/cloud.md [hunk 25]
```diff
@@ -1332,3 +1332,323 @@ For more information, see the [Python SDK Docs](https://python.viam.dev/autoapi/
...
+{{% /expand %}}
+
+For more information about managing permissions, see [Role-Based Access Control](/fleet/rbac/#permissions).
+
+### CreateKey
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+Create a new API key.
```
Comment (sguequierre) [inline docs/build/program/apis/cloud.md:1575]:
```suggestion
Create a new [API key](/fleet/rbac/#api-keys).
```
